org_fedora_oscap/content_discovery.py

```diff
@@ -112,23 +113,31 @@ def _fetch_files(self, ca_certs_path):
         return fetching_thread_name
 
     def _start_actual_fetch(self, ca_certs_path):
-        fetching_thread_name = None
+        fetching_thread_name = common.THREAD_FETCH_DATA
 
         scheme = self.content_uri.split("://")[0]
         if is_network(scheme):
-            fetching_thread_name = data_fetch.wait_and_fetch_net_data(
-                self.content_uri,
-                self.dest_file_name,
-                ca_certs_path
-            )
-        else:  # invalid schemes are handled down the road
-            fetching_thread_name = data_fetch.fetch_local_data(
-                self.content_uri,
-                self.dest_file_name,
-            )
+            try:
+                data_fetch.wait_for_network()
+            except common.OSCAPaddonNetworkError:
+                msg = _("Network connection needed to fetch data.")
+                raise common.OSCAPaddonNetworkError(msg)
+
+        fetch_data_thread = AnacondaThread(
+            name=fetching_thread_name,
+            target=self.fetch_operation,
+            args=(self.content_uri, self.dest_file_name, ca_certs_path),
+            fatal=False)
+
+        # register and run the thread
+        threadMgr.add(fetch_data_thread)
+
         return fetching_thread_name
 
-    def finish_content_fetch(self, fetching_thread_name, fingerprint):
+    def fetch_operation(self, uri, out_file, ca_certs_path=None):
+        return data_fetch.fetch_data(uri, out_file, ca_certs_path)
```

Comment:
The other function is somewhat coupled with the `data_fetch` module - the way it is unit-tested as well as the way exceptions it can raise are defined.
The whole data_fetch module could be merged into the bringer in the future, but I think that it is OK to have such thin wrapper.
The fetch operation as a separate method is convenient to be able to override it in child classes e.g. for testing purposes.